service/src/main/scala/org/apache/celeborn/server/common/http/api/CelebornOpenApiResource.scala

```diff
@@ -81,9 +82,10 @@ class CelebornOpenApiResource extends BaseOpenApiResource with ApiRequestContext
     }
   }
 
-  private def setCelebornOpenAPIDefinition(openApi: OpenAPI, requestBaseUrl: String): OpenAPI = {
-    // TODO: to improve when https is enabled.
-    val apiUrls = List(requestBaseUrl, s"http://${httpService.connectionUrl}/").distinct
+  private def setCelebornOpenAPIDefinition(openApi: OpenAPI, requestBaseUri: URI): OpenAPI = {
+    val httpScheme = if (httpService.httpSslEnabled()) "https:" else "http:"
+    val requestBaseUrl = s"$httpScheme${requestBaseUri.getSchemeSpecificPart}"
+    val apiUrls = List(requestBaseUrl, s"$httpScheme//${httpService.connectionUrl}/").distinct
```

Comment:
To fix the swagger request url scheme inconsistent when SSL enabled.
<img width="967" alt="image" src="https://github.com/user-attachments/assets/65e9f87a-a8ad-4621-86dc-58dc96ef9e98">

Reply:
```
scala> val uri = new java.net.URI("https://celeborn.vip.com:9098/").getSchemeSpecificPart
val uri: String = //celeborn.vip.com:9098/

scala> 
```